From: Egbert Eich <eich@suse.com>
Date: Mon Mar 28 08:14:52 2022 +0200
Subject: Use CC and full command line instead of hard-coding gcc for AVX512 checking
Patch-mainline: Not yet
Git-repo: https://github.com/xianyi/OpenBLAS
Git-commit: 0b69fa6ddf7fc6d92bc42ef085f39337a4489f3c
References: 

Hard-coding gcc may not provide incorrect results when a different compiler
for the target build is used. To remain in sync with the main call to c_check,
pass the full command line.

Signed-off-by: Egbert Eich <eich@suse.com>
Signed-off-by: Egbert Eich <eich@suse.de>
---
 Makefile.prebuild | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
diff --git a/Makefile.prebuild b/Makefile.prebuild
index 4dad74d6..5e8874d4 100644
--- a/Makefile.prebuild
+++ b/Makefile.prebuild
@@ -71,7 +71,7 @@ endif
 
 
 getarch : getarch.c cpuid.S dummy $(CPUIDEMU)
-	avx512=$$(perl c_check - - gcc | grep NO_AVX512); \
+	avx512=$$(perl c_check - - $(CC) $(TARGET_FLAGS) $(CFLAGS) | grep NO_AVX512); \
 	$(HOSTCC) $(HOST_CFLAGS) $(EXFLAGS) $${avx512:+-D$${avx512}} -o $(@F) getarch.c cpuid.S $(CPUIDEMU)
 
 getarch_2nd : getarch_2nd.c config.h dummy
